src/Core/Addon/Module/ModuleManager.php

```diff
@@ -286,19 +293,37 @@ public function install($source)
         }
 
         if ($this->moduleProvider->isInstalled($name)) {
-            if ($source !== null) {
-                return $this->upgrade($name, 'latest', $source);
-            }
+            return $this->upgrade($name, 'latest', $source);
+        }
 
-            // Module is already installed
-            return true;
+        // The module is not in PS modules folder and no ZIP is given as source
+        // We try to retrieve the module's zip from another module (marketplace)
+        if (!$this->moduleProvider->isOnDisk($name) && empty($source)) {
```

Comment:
Question: rather than choosing the path "$source is empty" can we provide a $source$ that would be marketplace?

Reply:
I'm afraid I don't understand your question
You mean get the source from marketplace before calling `install` ? or give `$source = 'marketplace'` ? I remember that the marketplace is a non-native module and may be missing

Reply:
Isn't it the purpose of this empty condition ? i.e. triggering an installed module (like the marketplace) that will download the missing source ?

Now this makes me curious :)
What should happen if 2 or more modules handle to the same hook `actionAdminModuleInstallRetrieveSource` ?